pkg/registry/seedmanagement/managedseed/strategy.go

```diff
@@ -0,0 +1,123 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managedseed
+
+import (
+	"context"
+
+	"github.com/gardener/gardener/pkg/api"
+	"github.com/gardener/gardener/pkg/apis/seedmanagement"
+	"github.com/gardener/gardener/pkg/apis/seedmanagement/validation"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/storage/names"
+)
+
+// Strategy defines the strategy for storing managedseeds.
+type Strategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// NewStrategy defines the storage strategy for ManagedSeeds.
+func NewStrategy() Strategy {
+	return Strategy{api.Scheme, names.SimpleNameGenerator}
+}
+
+// NamespaceScoped returns true if the object must be within a namespace.
+func (Strategy) NamespaceScoped() bool {
+	return true
+}
+
+// PrepareForCreate mutates some fields in the object before it's created.
+func (s Strategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+	managedSeed := obj.(*seedmanagement.ManagedSeed)
+
+	managedSeed.Generation = 1
+	managedSeed.Status = seedmanagement.ManagedSeedStatus{}
+}
+
+// PrepareForUpdate is invoked on update before validation to normalize
+// the object.  For example: remove fields that are not to be persisted,
+// sort order-insensitive list fields, etc.  This should not remove fields
+// whose presence would be considered a validation error.
+func (s Strategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newManagedSeed := obj.(*seedmanagement.ManagedSeed)
+	oldManagedSeed := old.(*seedmanagement.ManagedSeed)
+	newManagedSeed.Status = oldManagedSeed.Status
+
+	if !apiequality.Semantic.DeepEqual(oldManagedSeed.Spec, newManagedSeed.Spec) {
+		newManagedSeed.Generation = oldManagedSeed.Generation + 1
+	}
+}
+
+// Validate validates the given object.
+func (Strategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
+	managedSeed := obj.(*seedmanagement.ManagedSeed)
+	return validation.ValidateManagedSeed(managedSeed)
+}
+
+// Canonicalize allows an object to be mutated into a canonical form. This
+// ensures that code that operates on these objects can rely on the common
+// form for things like comparison.  Canonicalize is invoked after
+// validation has succeeded but before the object has been persisted.
+// This method may mutate the object.
+func (Strategy) Canonicalize(obj runtime.Object) {
+}
+
+// AllowCreateOnUpdate returns true if the object can be created by a PUT.
+func (Strategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+// AllowUnconditionalUpdate returns true if the object can be updated
+// unconditionally (irrespective of the latest resource version), when
+// there is no resource version specified in the object.
+func (Strategy) AllowUnconditionalUpdate() bool {
+	return true
```

Comment:
I assume, this should be false, otherwise concurrency control via the `resourceVersion` will be disabled.
```suggestion
	return false
```

I see, we have it set to `false` also for some other strategies. This seems wrong, or am I missing something here?

Reply:
You mean set to `true`, as in https://github.com/gardener/gardener/blob/5a58c19bcd708bb1566e494cb7f08d479bc3cd08/pkg/registry/core/seed/strategy.go#L93-L95 (the place I copy/pasted from). Yes I guess this is wrong, should we try to improve it?

Reply:
Ah yes, I meant `true`.
I see, it has been there already since the initial commit of gardener 😱 
@rfranzke any reason for having `AllowUnconditionalUpdate` return `true` for some of gardener's resources?

Reply:
We decided to set this as `false` (don't allow unconditional updates).